onmt/model_builder.py

```diff
@@ -191,40 +213,62 @@ def build_base_model(model_opt, fields, gpu, checkpoint=None, gpu_id=None):
         pad_idx = tgt_base_field.vocab.stoi[tgt_base_field.pad_token]
         generator = CopyGenerator(model_opt.dec_rnn_size, vocab_size, pad_idx)
 
+    if model_opt.is_bert:
+        model = encoder
```

Comment:
Why the encoder becomes the whole model? If there is no decoder, then how can I use the model for machine translation?

Reply:
BERT is Bidirectional Encoder Representations from Transformers. It is not a machine translation model.
https://arxiv.org/abs/1810.04805

Reply:
I know this :) But this is a PR to a machine translation system. If the machine translation cannot benefit from this PR then why bother? If you want use BERT for basic downstream tasks (e.g. classification), then you can use Hugging Face's Transformers.

Reply:
OpenNMT-py is not only machine translation, there are other tasks such as Speech2Text, Image2Text, Vid2Text, etc.
This PR is meant to add BERT task(s) and components to the framework to allow users to experiment. Nothing stops you to try and build a custom model using BERT pre-trained encoder layers and standard Transformer decoder layers for instance, but I think it has been tried without great benefits.

Reply:
Hmm, but AFAIK, all the tasks you mentioned (i.e. Speech2Text, Image2Text, Video2Text) are formalised as transduction problems and, therefore, require some sort of a decoder.

Reply:
Sure. All those things are not mutually exclusive.
Feel free to share any research on [the forum](http://forum.opennmt.net/) and eventually PR if you have anything interesting regarding using BERT in a seq2seq task.